src/Azure.DataApiBuilder.Mcp/BuiltInTools/CreateRecordTool.cs

```diff
@@ -37,51 +52,235 @@ public Tool GetToolMetadata()
+                        { 
+                            Type = "text", 
+                            Text = $"Successfully created record in entity '{entityName}'. Result: {JsonSerializer.Serialize(createdResult.Value)}"
+                        }]
+                    };
+                }
+                else if (result is ObjectResult objectResult)
+                {
+                    return new CallToolResult
+                    {
+                        Content = [new TextContentBlock 
+                        { 
+                            Type = "text", 
+                            Text = $"Record creation completed with status {objectResult.StatusCode}. Result: {JsonSerializer.Serialize(objectResult.Value)}"
+                        }]
+                    };
+                }
+                else
+                {
+                    return new CallToolResult
+                    {
+                        Content = [new TextContentBlock { Type = "text", Text = $"Successfully created record in entity '{entityName}'" }]
+                    };
+                }
             }
             catch (Exception ex)
             {
-                return Task.FromResult(new CallToolResult
+                return new CallToolResult
                 {
                     Content = [new TextContentBlock { Type = "text", Text = $"Error: {ex.Message}" }]
-                });
+                };
             }
         }
+
+        private static bool TryResolveAuthorizedRole(
+        HttpContext httpContext,
+        IAuthorizationResolver authorizationResolver,
+        string entityName,
+        out string error)
+        {
+            error = string.Empty;
+
+            string roleHeader = httpContext.Request.Headers[AuthorizationResolver.CLIENT_ROLE_HEADER].ToString();
+
+            if (string.IsNullOrWhiteSpace(roleHeader))
+            {
+                error = "Client role header is missing or empty.";
+                return false;
+            }
+
+            string[] roles = roleHeader
+                .Split(',', StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries)
+                .Distinct(StringComparer.OrdinalIgnoreCase)
+                .ToArray();
+
+            if (roles.Length == 0)
+            {
+                error = "Client role header is missing or empty.";
+                return false;
+            }
+
+            foreach (string role in roles)
+            {
+                bool allowed = authorizationResolver.AreRoleAndOperationDefinedForEntity(
+                    entityName, role, EntityActionOperation.Create);
+
+                if (allowed)
+                {
+                    return true;
+                }
+            }
+
+            error = "You do not have permission to create records for this entity.";
+            return false;
+        }
+
+        private static CallToolResult BuildErrorResult(
+        string errorType,
```

Comment:
nit: should be tabs before each argument declaration - similar to other function definitions.